components/eventsrv-protocol/protocols/event.proto

```diff
@@ -81,44 +81,43 @@ message CensusEntry {
   optional SysInfo sys = 6;
   /// Package information of service
   optional PackageIdent pkg = 7;
+  /// `true` if the service has successfully initialized
+  optional bool initialized = 8;
+  /// Depot that is checked for updates
+  optional string depot_url = 9;
+  /// Chennel chat is checked for updates
+  optional string channel = 10;
+  /// Persistent or transient
+  optional string start_style = 11;
+  /// Topology used by the service
+  optional string topology = 12;
+  /// Update strategy used by the service
+  optional string update_strategy = 13;
+  /// Service group application name
+  optional string application = 14;
+  /// Service group environment name
+  optional string environment = 15;
   /// `true` if this service instance is the leader when in a leader topology
-  optional bool leader = 8;
+  optional bool leader = 16;
```

Comment:
I was told once upon a time that new fields should always be added at the end of a protobuf message because you don't want to change the existing field order.  Is that still the case or did that change?

Reply:
It's not that the physical order in the definition is important, it's the field numbers themselves. If those change then I think you get into situations where you lose your backward-compatibility.

TL;DR: `leader` should stay as field 8.

Reply:
Since the very message itself is changing from a CensusEntry to a ServiceUpdate I did not consider compatibility since that alone will break it. @reset and I discussed that a breaking change is acceptable here.

Reply:
Ah, so it is! Yeah, this is fine; sorry!

Reply:
Ah, thanks for clarifying.  Makes sense.

Reply:
Thank you all for being on guard about this! In this case, yeah, it's a do-over so we can break :)